UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java

```diff
@@ -228,19 +250,19 @@ void notConnected() {
     void drive(int throttleSpeed, int steeringAngle, String actionDescription) {
         notConnected();
         Log.i(TAG, actionDescription);
-        if(throttleSpeed > speed + 5 || throttleSpeed < speed - 5 || throttleSpeed == 0){
-            speed = throttleSpeed;
+        if(throttleSpeed > lastSentSpeed + 5 || throttleSpeed < lastSentSpeed - 5 || throttleSpeed == 0){
+            lastSentSpeed = throttleSpeed;
             mMqttClient.publish(THROTTLE_CONTROL, Integer.toString(throttleSpeed), QOS, null);
         }
-        if(steeringAngle > 10 && angle <= 0){
-            angle = 30;
-            mMqttClient.publish(STEERING_CONTROL, Integer.toString(angle), QOS, null);
-        }else if(steeringAngle < -10 && angle >= 0){
-            angle = -30;
-            mMqttClient.publish(STEERING_CONTROL, Integer.toString(angle), QOS, null);
-        }else if (steeringAngle <= 10 && steeringAngle >= -10 && angle != 0){
-            angle = 0;
-            mMqttClient.publish(STEERING_CONTROL, Integer.toString(angle), QOS, null);
+        if(steeringAngle > 10 && lastSentAngle <= 0){
+            lastSentAngle = 30;
+            mMqttClient.publish(STEERING_CONTROL, Integer.toString(lastSentAngle), QOS, null);
+        }else if(steeringAngle < -10 && lastSentAngle >= 0){
+            lastSentAngle = -30;
+            mMqttClient.publish(STEERING_CONTROL, Integer.toString(lastSentAngle), QOS, null);
+        }else if (steeringAngle <= 10 && steeringAngle >= -10 && lastSentAngle != 0){
+            lastSentAngle = 0;
+            mMqttClient.publish(STEERING_CONTROL, Integer.toString(lastSentAngle), QOS, null);
         }
```

Comment:
This also resolves #58. I'd like a bit more clarification as to why we have double backed on sending messages in quanta (aka minimum required amount). When I pitched this idea during Sprint 2, it was shut down for reasons of "accuracy"; that it would be best to constantly be sending messages to the car. What happed to that requirement of constantly send messages to the car for every change of state, regardless of how small?

Reply:
Decreasing the accuracy of the joystick was not in our plans until we made sure that the camera had little impact on the latency. Currently, this code has been implemented mainly as a way to be able to test the functionalities, by publishing messages only if different from the previous ones within a specified range.

The branch "bug_motion_accuracy"  is dedicated for the code and documentation that will resolve #58 .

Reply:
I understand. Great job!